src/components/fullscreen-control.js

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {document} from '../utils/globals';
+import PropTypes from 'prop-types';
+import BaseControl from './base-control';
+import {createElement} from 'react';
+import {FullscreenControl as MapboxFullscreenControl} from '../utils/mapboxgl';
+
+const propTypes = Object.assign({}, BaseControl.propTypes, {
+  // Custom className
+  className: PropTypes.string,
+  /* eslint-disable max-len */
+  // `container` is the [compatible DOM element](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#Compatible_elements)
+  // which should be made full screen. By default, the map container element will be made full screen.
+  /* eslint-enable max-len */
+  container: PropTypes.object
+});
+
+const defaultProps = Object.assign({}, BaseControl.defaultProps, {
+  className: '',
+  container: null
+});
+
+export default class FullscreenControl extends BaseControl {
+
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+
+  constructor(props) {
+    super(props);
+
+    this._mapboxFullscreenControl = null;
+
+    this.state = {
+      isFullscreen: false
+    };
+  }
+
+  componentDidMount() {
+    const container = this.props.container || this._context.mapContainer;
```

Comment:
`StaticMap` can be used standalone. `context.mapContainer` will be empty in that case.

Reply:
My understanding is `StaticMap` should also support `FullscreenControl`. Then I need find a way to get the `MapContainer`(container also include the controls) in standalone `StaticMap` case.